src/port/port.tcl

```diff
@@ -4577,6 +4577,23 @@ proc parse_options { action ui_options_name global_options_name } {
                         # debug implies verbose
                         set ui_options(ports_verbose) yes
                     }
+                    [0-3] {
+                        #d[0-3]
+                        # TEMP HACK: Since args processed char-by-char, just use single-digit
+                        # numeric arg for now. To fix, we either need to pre-process the args,
+                        # and ensure 'dX' is combined before this switch. Or use non-destructive
+                        # lookahead.
```

Comment:
Well, the problem with processing args one character at a time, without any context/state, is that they can be repeated ad infinitum. As it stands today, without this patch, the following veriations are valid. Or at least, we don't validate whether they make sense or not:

```
$ port -dddddddddddddddddddddddddddddd info my_port
$ port -d -d -d -d -d -d -d -d -d -d info my_port
```

You get the picture.

I'd prefer that we not allow that, actually. As well as preprocess the args for cases like the debug level. (And combine multi-character args - like `dX` - before passing them through the switch statement.)

Reply:
As for repeating `-d` multiple times, to specify the level, hmmm.

Yes, that paradigm is sometimes used/allowed in command-line utilities. But it seems clearer to simply specify -dX, with the numeric level. This would be even more important if we allowed levels beyond debug3, as it gets a bit ridiculous after a while.

I mean, we'd have to specify `-dddd` for debug3? Yeah, no thanks. It's difficult to read, and you have to carefully count the characters. Perhaps it's usable when maximizing out at three levels, but definitely wouldn't be my choice.

Reply:
Sorry, I wasn't clear. Yes, I meant the `port -ddd` repetition as a way to specify the debug level. I would prefer this over an option with a digit argument.

Having a `-d[X]` that can be used with or without an argument is a bad choice. If an option does not take an argument, then it can be combined freely with any other option of that kind into a single `-abc...`. If an option takes an argument, then that follows in the next argument, not immediately after the letter. These rules are simple, but "`-d` can be combined with other options, but if the next letter is a digit, then it is treated differently" complicates both parsing and documentation. For example, `-dyp` is equivalent to `-d -y -p`, but `-d3yp` would be `-d3 -y -p`.

For the proposed `-dX` you would have to look at the argument that follows and decide whether this is valid or not and then treat it differently. This mixes the syntax parsing of options with their semantics and I do not like that. It will only work for this one special case where `-d` takes a single digit. I do not see how allowing this would help with any other option in the future as it cannot be generalized. It also forbids `-0` to `-9` from ever being used as an option, as it would be ambiguous.

Therefore the only choice would be `-d X` and make it *always* take this argument. However, as I wouldn't want to change the existing `-d`, we would even have to settle on a different letter to set the debug log level... And then we end up with both `-d` and `-L 3` (or whatever).

Reply:
Rainer, I'm very open to whatever you folks want to use, in terms of arguments. If that's what it takes to get this merged, I'll implement whatever argument style you folks wish!

Repetition is certainly fine. The single-digit approach we're using now works well too.

Tell me what you folks prefer, and I'll do it!

Reply:
In other words, "What can I do to get you to buy this new car TODAY?"  :-)